### ISSUE TITLE
handle-callback-err shouldn't trigger when `arguments` is used

### ISSUE THREAD

**Issue by Fishrock123** [eslint/eslint#11031]
**What rule do you want to change?**
`handle-callback-err`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
IMO the default behavior should be changed, maybe an option could be added.

**Please provide some example code that this change will affect:**

Saw some code trigger this one time that looked something like this: (I can't find the exact source now..)

```js
something(function (err /* err is ... */) {
  this.emit('some-event', arguments)
})
```

Or at least, something _like_ that.

**What does the rule currently do for this code?**
It says `err` is not handled, but it actually is.

**What will the rule do after it's changed?**
See that `arguments` was used, and not fail the rule check.

I'd also like it to be able to check for `arguments[0]` specifically and not other indexes... if possible.

**Are you willing to submit a pull request to implement this change?**
Yes, but I am fairly unfamiliar with eslint rule programming so I'm having a hard time figuring out how to do the `arguments` check.

**Comment by nzakas** [comment]
Thanks for the issue. The rule is behaving correctly because `err` is not being used in your example. If you aren't going to use `err`, than I'd recommend removing it from the function declaration to avoid triggering this error. I'm not in favor of making this change, but let's see if any other team member feels differently.

**Comment by platinumazure** [comment]
With the advent of rest parameters, I feel this could be handled differently:

```js
something(function (err, ...args) {
  this.emit('some-event', err, ...args)
})
```

This ensures explicit usage of err, without requiring all other potential arguments to be specified.

Naturally, some users have to support old runtimes, so this won't work for everyone. But I think it's a much better solution overall, and I'm not sure it's worth increasing the complexity of this rule for what might be a minority of our users.

**Comment by Fishrock123** [comment]
I'm using eslint against foreign sources, so I don't have any say about the source code in question here.

I'd hazard a guess that's not recommended eslint usage, but it is certainly _a_ use-case 🤷‍♂️.

I can just modify the rule myself (I'm still a bit confounded as to how exactly I should modify it), but figured it might be upstream-able as it does seem realistic.
